src/Tearing/TearingEngine.inl

```diff
@@ -840,38 +813,28 @@ inline void TearingEngine<DataTypes>::calculate_inverse_distance_weights(std::ve
 
      helper::ReadAccessor< Data<VecCoord> > x(d_input_positions);
      
-     const Coord p = x[(int)vertex];
+     const Coord p = x[vertexId];
 
-    for (unsigned int tri=0 ; tri< ValidTrianglesAround.size(); tri++)
+    for (TriangleID valTriId : ValidTrianglesAround)
```

Comment:
In this for loop, if you put "auto" instead of TriangleID, is it still correct? How about "index" instead of TriangleID?

Reply:
auto is totally correct.
index would work also but we come back to previous comment.